File: CS/CustomDrawHolidayCells/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;

namespace CustomDrawHolidayCells {
	public partial class Form1 : Form {
		public Form1() {
			InitializeComponent();

			AddHoliday(DateTime.Today);
		}
		private WorkDaysCollection Holidays { get { return schedulerControl1.WorkDays; } }

		private void btnAddHoliday_Click(object sender, EventArgs e) {
			DayIntervalCollection days = new DayIntervalCollection();
			days.Add(schedulerControl1.SelectedInterval);

			schedulerControl1.BeginUpdate();
			try {
				for (int i = 0; i < days.Count; i++)
					AddHoliday(days[i].Start);
			} finally {
				schedulerControl1.EndUpdate();
			}
			 
		}
		private void AddHoliday(DateTime holidayDate) {
			TimeInterval interval = new TimeInterval(holidayDate, TimeSpan.FromDays(1));
			if (!IsHoliday(interval))
				Holidays.AddHoliday(holidayDate, string.Format("Holiday-{0}", holidayDate.ToShortDateString()));
		}
		private bool IsHoliday(TimeInterval interval) {
			int count = Holidays.Count;
			for (int i = 0; i < count; i++) {
				Holiday holiday = Holidays[i] as Holiday;
				if (holiday != null) {
					TimeInterval holidayInterval = new TimeInterval(holiday.Date, TimeSpan.FromDays(1));
					if (holidayInterval.IntersectsWithExcludingBounds(interval))
						return true;
				}
			}
			return false;
		}
		private void schedulerControl1_CustomDrawTimeCell(object sender, CustomDrawObjectEventArgs e) {
			SchedulerViewCellBase cell = e.ObjectInfo as SchedulerViewCellBase;
			if (cell != null && IsHoliday(cell.Interval)) {
				cell.Appearance.BackColor = Color.Tomato;
			}
		}

	}
}

File: VB/CustomDrawHolidayCells/Form1.vb
Imports DevExpress.Schedule
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Namespace CustomDrawHolidayCells
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()

			AddHoliday(DateTime.Today)
		End Sub
		Private ReadOnly Property Holidays() As WorkDaysCollection
			Get
				Return schedulerControl1.WorkDays
			End Get
		End Property

		Private Sub btnAddHoliday_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnAddHoliday.Click
			Dim days As New DayIntervalCollection()
			days.Add(schedulerControl1.SelectedInterval)

			schedulerControl1.BeginUpdate()
			Try
				For i As Integer = 0 To days.Count - 1
					AddHoliday(days(i).Start)
				Next i
			Finally
				schedulerControl1.EndUpdate()
			End Try

		End Sub
		Private Sub AddHoliday(ByVal holidayDate As DateTime)
			Dim interval As New TimeInterval(holidayDate, TimeSpan.FromDays(1))
			If Not IsHoliday(interval) Then
				Holidays.AddHoliday(holidayDate, String.Format("Holiday-{0}", holidayDate.ToShortDateString()))
			End If
		End Sub
		Private Function IsHoliday(ByVal interval As TimeInterval) As Boolean
			Dim count As Integer = Holidays.Count
			For i As Integer = 0 To count - 1
				Dim holiday As Holiday = TryCast(Holidays(i), Holiday)
				If holiday IsNot Nothing Then
					Dim holidayInterval As New TimeInterval(holiday.Date, TimeSpan.FromDays(1))
					If holidayInterval.IntersectsWithExcludingBounds(interval) Then
						Return True
					End If
				End If
			Next i
			Return False
		End Function
		Private Sub schedulerControl1_CustomDrawTimeCell(ByVal sender As Object, ByVal e As CustomDrawObjectEventArgs) Handles schedulerControl1.CustomDrawTimeCell
			Dim cell As SchedulerViewCellBase = TryCast(e.ObjectInfo, SchedulerViewCellBase)
			If cell IsNot Nothing AndAlso IsHoliday(cell.Interval) Then
				cell.Appearance.BackColor = Color.Tomato
			End If
		End Sub

	End Class
End Namespace